File: python/src/mpeg_o/axis_descriptor.py
"""``AxisDescriptor`` — names + units + sampling metadata for one signal axis."""
from __future__ import annotations

from dataclasses import dataclass

from .enums import SamplingMode
from .value_range import ValueRange


@dataclass(frozen=True, slots=True)
class AxisDescriptor:
    """Describes a single axis of a :class:`~mpeg_o.signal_array.SignalArray`.

    Immutable value class. ``name`` is a short token (``"mz"``,
    ``"chemical_shift"``, ...) and ``unit`` is a free-form label
    (``"m/z"``, ``"ppm"``, ``"s"``, ...).

    Parameters
    ----------
    name : str
        Semantic name of the axis.
    unit : str
        Unit of the axis values.
    value_range : ValueRange or None, default None
        Numeric bounds of the axis. ``None`` when unknown.
    sampling_mode : SamplingMode, default SamplingMode.UNIFORM
        Whether samples are regularly spaced.

    Notes
    -----
    API status: Stable.

    Cross-language equivalents
    --------------------------
    Objective-C: ``MPGOAxisDescriptor`` · Java:
    ``com.dtwthalion.mpgo.AxisDescriptor``.
    """

    name: str
    unit: str
    value_range: ValueRange | None = None
    sampling_mode: SamplingMode = SamplingMode.UNIFORM

File: java/src/main/java/com/dtwthalion/mpgo/AxisDescriptor.java
/*
 * MPEG-O Java Implementation
 * Copyright (C) 2026 DTW-Thalion
 * SPDX-License-Identifier: LGPL-3.0-or-later
 */
package com.dtwthalion.mpgo;

import com.dtwthalion.mpgo.Enums.SamplingMode;

/**
 * Describes a single axis of a {@link SignalArray}: its semantic name,
 * unit, numeric range, and sampling mode. Immutable value class.
 *
 * <p><b>API status:</b> Stable.</p>
 *
 * <p><b>Cross-language equivalents:</b> Objective-C
 * {@code MPGOAxisDescriptor}, Python
 * {@code mpeg_o.axis_descriptor.AxisDescriptor}.</p>
 *
 * @param name          Semantic name (e.g. {@code "mz"}).
 * @param unit          Unit label (e.g. {@code "m/z"}).
 * @param valueRange    Numeric bounds, or {@code null} when unknown.
 * @param samplingMode  Whether samples are regularly spaced.
 * @since 0.6
 */
public record AxisDescriptor(
    String name,
    String unit,
    ValueRange valueRange,
    SamplingMode samplingMode
) {}
